packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Generator.CSharp.Expressions;
+
+namespace Microsoft.Generator.CSharp.Statements
+{
+    public sealed class CatchStatement : MethodBodyStatement, IEnumerable<MethodBodyStatement>
+    {
+        private readonly List<MethodBodyStatement> _body = [];
+        public ValueExpression? Exception { get; }
+        public IReadOnlyList<MethodBodyStatement> Body => _body;
+        public CatchStatement(ValueExpression? exception)
+        {
+            Exception = exception;
+        }
+
+        internal override void Write(CodeWriter writer)
+        {
+            writer.AppendRaw("catch");
+            if (Exception != null)
+            {
+                writer.AppendRaw(" (");
+                Exception.Write(writer);
+                writer.WriteRawLine(")");
+            }
+
+            using (writer.Scope())
+            {
+                foreach (var statement in Body)
+                {
+                    statement.Write(writer);
+                }
+            }
+        }
+
+        public CatchStatement Add(MethodBodyStatement statement)
```

Comment:
I think we did this for consistency with If?  I am not sure I am fond of that pattern can we leave these off here and we can consider flipping the consistency the other way?

Reply:
nvm, I see the reason we need this

Reply:
Needed to be able to use the list initializer.